src/periodic.jl

```diff
@@ -0,0 +1,29 @@
+function PeriodicCallback(f, Δt::Number; kwargs...)
+    @assert Δt > 0
+
+    # Value of `t` at which `f` should be called next:
+    t_next = Ref(typemax(Δt))
+    condition = (t, u, integrator) -> t == t_next[]
+
+    # Call f, update t_next, and make sure we stop at the new t_next
+    affect! = function (integrator)
+        f(integrator)
+
+        # Schedule next call to `f` using `add_tstops!`, but be careful not to keep integrating forever
+        t_new = t_next[] + Δt
+        if any(t -> t > t_new, integrator.opts.tstops.valtree) # TODO: accessing internal data...
```

Comment:
yeah, there must be a better way to iterate through a heap

Reply:
Yeah, I guess it would at least be better to reverse-iterate.

Could also:
* store the maximum time in the integrator options, in addition to `tstops` (although it then becomes possible to have bugs where `tstops` is out of sync with that maximum time).
* switch to a min-max heap (not yet implemented in DataStructures.jl and perhaps overkill)